src/main.rs

```diff
@@ -154,10 +202,20 @@ fn run() -> Result<ExitCode> {
         None
     };
 
-    let command = if let Some(args) = matches.values_of("exec") {
-        Some(CommandTemplate::new(args))
-    } else if let Some(args) = matches.values_of("exec-batch") {
-        Some(CommandTemplate::new_batch(args)?)
+    let mut commands: Vec<CommandTemplate> = vec![];
+
+    if matches.is_present("exec") {
+        let ranges = ContinuousRangeIterator::new(matches.indices_of("exec").unwrap().peekable());
+        let args: Vec<&str> = matches.values_of("exec").unwrap().collect();
+        for (start, end) in ranges {
+            commands.push(CommandTemplate::new(&args[start..end]));
+        }
```

Comment:
I wanted to call `.take(n)` in a loop on the value iterator, but calling take moves the iterator. Since that didn't work, I tried having the iterator return slices of some internal value array, which also made the borrow checker sad. I gave up and just  built a vec out of the values. If there is some way to have the Iterator own the values, and return references to those values, I would like to know it.

Reply:
I think you need to store the values somehow as long as you have to pass something like `&args[start..end]` to `CommandTemplate::new` if the only thing you have is the `Iterator` over single values. An alternative might be to rewrite `CommandTemplate::new` to take an iterator. This way, you could maybe pass `X.take(n)` to `CommandTemplate::new(…)`.

(Maybe it would also help to provide both indices and the values to `ContinuousRangeIterator`?)